www/js/lib/zimArchive.js

```diff
@@ -668,6 +668,25 @@ ZIMArchive.prototype.getMetadata = function (key, callback) {
     });
 };
 
+/**
+ * Get all Metadata required for ZIM file security confirmation
+ * @returns {Promise<Object>} A Promise that resolves with the metadata object (contains Name, Creator, Publisher, Scraper)
+ */
+ZIMArchive.prototype.getConfirmationMetadata = function () {
+    var that = this;
+    return new Promise(function (resolve, reject) {
+        that.getMetadata('Name', function (name) {
+            that.getMetadata('Creator', function (creator) {
+                that.getMetadata('Publisher', function (publisher) {
+                    that.getMetadata('Scraper', function (scraper) {
+                        resolve({ name: name, creator: creator, publisher: publisher, scraper: scraper });
+                    });
+                });
+            });
+        });
+    })
+}
+
```

Comment:
It's good that you worked out how to get the metadata! However, it's reinventing the wheel, because the current code has two places to get metadata: the first is while the ZIM is loading, and the second is asynchronously after the ZIM has loaded (for non-critical data). All you need to do is add the time-critical metadata (i.e. these keys) to the list here: https://github.com/kiwix/kiwix-js/blob/main/www/js/lib/zimArchive.js#L82, and remove any of these keys that appear in the list of non-critical data here: https://github.com/kiwix/kiwix-js/blob/main/www/js/lib/zimArchive.js#L91. Also, JFYI, the more elegant way to gather a series of Promises is to use `Promise.all()` as is used in those functions, rather than a chain of Promises as you do here. Once you've modified the code around line 82, you can remove this function. The metadata you need will then be available in `archive.file.xxxx` in `archiveReadyCallback` (or `selectedArchive.file.xxxx` in the rest of `app.js`).